yarn-project/sequencer-client/src/publisher/l1-publisher.ts

```diff
@@ -707,32 +711,24 @@ export class L1Publisher {
       const proofHex: Hex = `0x${args.proof.withoutPublicInputs().toString('hex')}`;
       const txArgs = [...this.getSubmitEpochProofArgs(args), proofHex] as const;
       this.log.info(`SubmitEpochProof proofSize=${args.proof.withoutPublicInputs().length} bytes`);
-      await this.rollupContract.simulate.submitEpochRootProof(txArgs, { account: this.account });
-      return await this.rollupContract.write.submitEpochRootProof(txArgs, { account: this.account });
+
+      const txReceipt = await this.l1TxUtils.sendAndMonitorTransaction({
+        to: this.rollupContract.address,
+        data: encodeFunctionData({
+          abi: this.rollupContract.abi,
+          functionName: 'submitEpochRootProof',
+          args: txArgs,
+        }),
+      });
+
+      return txReceipt.transactionHash;
     } catch (err) {
       this.log.error(`Rollup submit epoch proof failed`, err);
       return undefined;
     }
   }
 
-  private async prepareProposeTx(encodedData: L1ProcessArgs, gasGuess: bigint) {
-    // We have to jump a few hoops because viem is not happy around estimating gas for view functions
-    const computeTxsEffectsHashGas = await this.publicClient.estimateGas({
-      to: this.rollupContract.address,
-      data: encodeFunctionData({
-        abi: this.rollupContract.abi,
-        functionName: 'computeTxsEffectsHash',
-        args: [`0x${encodedData.body.toString('hex')}`],
-      }),
-    });
-
-    // @note  We perform this guesstimate instead of the usual `gasEstimate` since
-    //        viem will use the current state to simulate against, which means that
-    //        we will fail estimation in the case where we are simulating for the
-    //        first ethereum block within our slot (as current time is not in the
-    //        slot yet).
-    const gasGuesstimate = computeTxsEffectsHashGas + gasGuess;
```

Comment:
I've dded an option to add a fixed buffer when sending a tx so we're still adding the same amount to whatever estimate the tx gets.

Reply:
But the catch here is that the gas estimation is done on a _different_ function than the one actually called in the transaction, because the called one would revert.

Reply:
huh so it should be failing? I haven't seen this revert at all 🤔

Reply:
It should, yeah. Maybe it's working just because of timings, things being slow enough that we don't end up on the first ethereum block?